File: frontend/src/app/cad/infrastructure/ocorrencia.response.ts
import { Ocorrencia } from '../domain/ocorrencia.model';

export interface OcorrenciaResponse {
  id: string;
  protocolo: string;
  natureza: string;
  gravidade: string;
  inicioAtendimento: string;
  telefone: string | null;
  pabxUid: string | null;
  endereco: string;
  latitude: number;
  longitude: number;
  situacao: string;
  mesa: string | null;
  encaminhadaEm: string | null;
  prefixoEmpenhado: string | null;
  inicioDeslocamento: string | null;
  noLocalEm: string | null;
  noLocalManual: boolean;
  encerradaEm: string | null;
}

export function paraOcorrencia(response: OcorrenciaResponse): Ocorrencia {
  return {
    id: response.id,
    protocolo: response.protocolo,
    natureza: response.natureza,
    gravidade: response.gravidade,
    inicioAtendimento: response.inicioAtendimento,
    telefone: response.telefone,
    pabxUid: response.pabxUid,
    endereco: response.endereco,
    latitude: response.latitude,
    longitude: response.longitude,
    situacao: response.situacao,
    mesa: response.mesa,
    encaminhadaEm: response.encaminhadaEm,
    prefixoEmpenhado: response.prefixoEmpenhado ?? null,
    inicioDeslocamento: response.inicioDeslocamento ?? null,
    noLocalEm: response.noLocalEm ?? null,
    noLocalManual: response.noLocalManual ?? false,
    encerradaEm: response.encerradaEm ?? null,
  };
}

File: backend/infrastructure/src/main/java/br/gov/mt/sesp/cicc/infrastructure/rest/OcorrenciaResponse.java
package br.gov.mt.sesp.cicc.infrastructure.rest;

import br.gov.mt.sesp.cicc.application.cad.OcorrenciaVista;

import java.time.Instant;

public record OcorrenciaResponse(
        String id,
        String protocolo,
        String natureza,
        String gravidade,
        Instant inicioAtendimento,
        String telefone,
        String pabxUid,
        String endereco,
        String pontoReferencia,
        double latitude,
        double longitude,
        String situacao,
        String mesa,
        Instant encaminhadaEm,
        String prefixoEmpenhado,
        Instant inicioDeslocamento,
        Instant noLocalEm,
        boolean noLocalManual,
        Instant encerradaEm
) {

    public static OcorrenciaResponse from(final OcorrenciaVista output) {
        return new OcorrenciaResponse(
                output.id(),
                output.protocolo(),
                output.natureza(),
                output.gravidade(),
                output.inicioAtendimento(),
                output.telefone(),
                output.pabxUid(),
                output.endereco(),
                output.pontoReferencia(),
                output.latitude(),
                output.longitude(),
                output.situacao(),
                output.mesa(),
                output.encaminhadaEm(),
                output.prefixoEmpenhado(),
                output.inicioDeslocamento(),
                output.noLocalEm(),
                output.noLocalManual(),
                output.encerradaEm()
        );
    }
}
